gconf/model.go

```diff
@@ -0,0 +1,107 @@
+package gconf
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*Conf)(nil)
+
+func (c *Conf) Validate() error {
+	return nil
+}
+
+func (c *Conf) Copy() orm.CloneableData {
+	// TODO
+	return c
+}
+
+type ConfBucket struct {
+	orm.Bucket
+}
+
+func NewConfBucket() *ConfBucket {
+	return &ConfBucket{
+		Bucket: orm.NewBucket("gconf", orm.NewSimpleObj(nil, &Conf{})),
+	}
+}
+
+func NewConf(propName []byte, value interface{}) (orm.Object, error) {
+	var conf Conf
+
+	switch v := value.(type) {
+	case string:
+		conf.Value = &Conf_String_{String_: v}
+	case int64:
+		conf.Value = &Conf_Int{Int: v}
+	case []byte:
+		conf.Value = &Conf_Bytes{Bytes: v}
+	case weave.Address:
+		conf.Value = &Conf_Bytes{Bytes: v}
+	case *coin.Coin:
+		conf.Value = &Conf_Coin{Coin: v}
+	case coin.Coin:
+		conf.Value = &Conf_Coin{Coin: &v}
+	default:
+		return nil, errors.Wrapf(errors.ErrInvalidType, "%T", value)
+	}
+
+	return orm.NewSimpleObj(propName, &conf), nil
+}
+
+func (b *ConfBucket) Load(db weave.KVStore, propName []byte, dest interface{}) error {
```

Comment:
Looks good, so it will load the protobuf, and then assert the caller passed in the proper type.
I like

Reply:
But... how about making propName a string? we only allow those, and then we can cast them to []byte in this package, rather than force the caller to do so